### DISCUSSION TITLE
Setting Timeout in v5

### DISCUSSION THREAD

**Discussion by thomasmcneill** [microsoftgraph/msgraph-sdk-dotnet#2726] [General]
I'm trying to find documentation on setting the timeout value with version 5.  Previous versions show being able to access the httpClient directly.  I've tried passing my own httpClient but then that breaks my client auth.  Can someone point me to some good articles on this subject with the latest version?  

My issue pertains to getting emails and expanding attachments.  I can get get most email attachments without issue but occasionally one breaks it and I get an exception for cancellation after the timeout.

**Comment by andrueastman** [comment]
Thanks for raising this @thomasmcneill 

Any chance you can share how you initializing your client? I believe with this constructor, you should still be able to pass your own client as well as an authprovider.

https://github.com/microsoftgraph/msgraph-sdk-dotnet/blob/cce3ac1f041ab07f126beaff4f8e40529f6394bf/src/Microsoft.Graph/GraphServiceClient.cs#L91

**Reply by thomasmcneill** [reply]
Nevermind, I think this is legimitely a service error.  I added a debug middleware, used the same query that gets the errors and token, it's one message, and I get 503 Service Unavailable in Postman as well.

**Comment by thomasmcneill** [comment]
Thanks for the response, I have tried the following and many variations of it.  With and without the handlers.  When I do this method the service fails with "One or more errors occurred. (Too many retries performed. More than 3 retries encountered while sending the request. (HTTP request failed with status code: ServiceUnavailable.) (HTTP request failed with status code: ServiceUnavailable.) (HTTP request failed..."

var clientSecretCredential = new ClientSecretCredential(tenantId, clientId, clientSecret, options);
var handlers = GraphClientFactory.CreateDefaultHandlers();
var httpClient = GraphClientFactory.Create(handlers);
 httpClient.Timeout = TimeSpan.FromMinutes(10);
var graphClient = new GraphServiceClient(httpClient, clientSecretCredential, scopes);
